### PR TITLE
[BUGFIX] `argilla`: normalize records when exporting flatten

### LINKED ISSUE #4936: [FEATURE] Review record flatten process
notes:

- user id -> username
- response status?
- "text" -> "fields.text" ?
- "metadata_key" -> "metadata.metadata_key"?

### PR DESCRIPTION
**Pull request by frascuchon**
# Pull Request Template

> [!NOTE]  
> This PR should be merged after merge https://github.com/argilla-io/argilla/pull/5137

This PR addresses problems when exporting records to dicts including records partially filled. This PR fixes the errors commented in https://github.com/argilla-io/argilla/issues/4936#issuecomment-2196890124

Close https://github.com/argilla-io/argilla/issues/4936

**Type of change**

- Bug fix (non-breaking change which fixes an issue)
- Refactor (change restructuring the codebase without changing functionality)
- Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- follows the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla/src/argilla/records/_io/_datasets.py [hunk 1]
```diff
@@ -64,13 +64,13 @@ def to_datasets(records: List["Record"]) -> "HFDataset":
...
-        record_dicts = GenericIO.to_list(records, flatten=True)
-        dataset = dataset_class.from_list(record_dicts)
+        record_dicts = GenericIO.to_dict(records, flatten=True)
+        dataset = dataset_class.from_dict(record_dicts)
         return dataset
 
     @staticmethod
     def _record_dicts_from_datasets(dataset: "HFDataset") -> List[Dict[str, Union[str, float, int, list]]]:
-        """Creates a dictionaries from a HF dataset that can be passed to DatasetRecords.add or DatasetRecords.update.
+        """Creates a dictionaries from an HF dataset that can be passed to DatasetRecords.add or DatasetRecords.update.
```
Comment (burtenshaw) [inline argilla/src/argilla/records/_io/_datasets.py]:
```suggestion
        """Creates a dictionaries from a HF dataset that can be passed to DatasetRecords.add or DatasetRecords.update.
```

File: argilla/tests/unit/test_io/test_hf_datasets.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+        ds = HFDatasetsIO.to_datasets(records)
+        assert ds.features == {
+            "_server_id": Value(dtype="null", id=None),
+            "id": Value(dtype="string", id=None),
+            "field": Value(dtype="string", id=None),
+            "other": Value(dtype="string", id=None),
+            "a": Value(dtype="string", id=None),
+            "b": Value(dtype="string", id=None),
+            "question.suggestion": Value(dtype="string", id=None),
```
Comment (burtenshaw) [inline argilla/tests/unit/test_io/test_hf_datasets.py:57]: How does this behave for spans, rating, and ranking? i.e. what type is the suggestion value cast to?

**Review by burtenshaw — APPROVED** [review]
Looks good. I would just like to understand the casting of question values to hf dataset features.

File: argilla/tests/unit/test_io/test_hf_datasets.py (hunk 2 above)
Comment (frascuchon) [reply argilla/tests/unit/test_io/test_hf_datasets.py:57]: I've include more value types in tests to cover and describe this.

**Pull request merged**